src/lib/types.rs

```diff
@@ -979,6 +979,85 @@ pub struct Task {
     pub mac: Option<PlatformOverrideTask>,
 }
 
+/// A dependency, defined either as a string or as a Dependency object
+#[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq)]
+#[serde(untagged)]
+pub enum StringTaskIdentifier {
```

Comment:
this is used for dependencies, maybe just call it DepedencyIdentifier?
regardless, i wouldn't write String in the name.